python/understack-flavor-matcher/flavor_matcher/flavor_spec.py

```diff
@@ -47,6 +51,24 @@ def stripped_name(self):
             raise Exception(f"Unable to strip envtype from flavor: {self.name}")
         return name
 
+    @property
+    def baremetal_nova_resource_class(self):
+        """Returns flavor name converted to be used with Nova flavor resources.
+
+        https://docs.openstack.org/ironic/latest/install/configure-nova-flavors.html
+        """
+        converted_name = re.sub(r"[^\w]", "", self.stripped_name).upper()
+        return f"RESOURCES:CUSTOM_BAREMETAL_{converted_name}"
```

Comment:
I believe `resources:` needs to be all lowercased, but the rest of it is uppercased.. I couldn't find it explicitly saying this in nova code ( https://github.com/search?q=repo%3Aopenstack%2Fnova+resources%3A+CUSTOM&type=code&p=1 ) but all their docs, examples, and tests have `resources:` lowercased. I was getting `nova.exception.NoValidHost: No valid host was found` with capitalized `RESOURCES:` but it started working when I changed the property name to lowercase `resources:`.

Reply:
Good catch, you are absolutely right - it [must be lowercased](https://opendev.org/openstack/nova/src/commit/e8a544f0e1134a4b260720cc634ad1997f936d6d/nova/scheduler/utils.py#L54). Fixed in the commit below - thanks